### DISCUSSION TITLE
Use an existing (static) public IP for emqx5 service

### DISCUSSION THREAD

**Discussion by klemen-df** [emqx/emqx-operator#669] [Q&A] [answered]
Hi,

we would like to use an already assign static public IP (in Azure).

One way is, of course, to create a new service (LoadBalancer type), select nodes through annotations and assign it the public static ip.

Is there any other way to achieve this functionality through the operator? To somehow point the operator towards an already existing public IP?

**Comment by Rory-Z** [comment]
Hi, @KlemenDanfoss Sorry for delay.
For now, the EMQX Operator doesn't support existing service resource, so deploy EMQX Custom Resource will definitely create a new service. 
As we know, the LoadBalancer service will create LB resource in Azure, so is any way band existing LB resource to k8s service by `metadata.annotation` ?  I'n not sure, I don't know much about Azure.

PS: We are trying to find a way to support external services. If we allow you to use an existing service to deploy EMQX CR, but this service will be updated by EMQX Operator with `.spec.port` and `.spec.selector`, do you think it is acceptable?

**Comment by klemen-df** [comment]
Hi,

I think we have two use cases here.

First one is that we just want to use our own (statically assigned) IP. If you check this guide ([click!](https://learn.microsoft.com/en-us/azure/aks/static-ip)) you can see that it's only a matter of providing the (wan) IP - kubernetes should take care of the rest. Of course one needs to make sure that that IP exists (not a task for the operator).

Second one is to use our own service entirely. For my use case (and probably broader) I would say that this is an acceptable solution (you proposed, with the selectors). 

Only drawback with proposals above is using IaaC. In terraform, for example, one would need to 'ignore_changes' - since changes would not come from terraform. But that's acceptable I would say.

**Reply by Rory-Z** [reply, marked as answer]
Hi, sorry for delay, and thanks for your feedback, this makes a lot of sense to me.

I read this guide ([click!](https://learn.microsoft.com/en-us/azure/aks/static-ip)), looks set `.metadata. annotations` and `.spec. loadBalancerIP` in EMQX CR's `serviceTemplate` it will ok.

Now it is work ?

**Comment by klemen-df** [comment]
Great, works!
